### PR TITLE
Oasis orientation

### PR DESCRIPTION
**Pull request by bingo619**

### CODE REVIEW

File: frontend/device/input.lua [hunk 1]
```diff
@@ -449,6 +458,31 @@ function Input:handleTouchEvPhoenix(ev)
...
+            if ev.value == DEVICE_ORIENTATION_PORTRAIT then
+                refreshUI = true
+            elseif ev.value == DEVICE_ORIENTATION_PORTRAIT_ROTATED then
+                refreshUI = true
+                rotation_mode = 2
+            end
+
+            if refreshUI then
+                local oldRotation = self.device.screen:getRotationMode()
+                self.device.screen:setRotationMode(rotation_mode)
```
Comment (houqp) [inline frontend/device/input.lua]: why not check `rotation_mode ~= oldRotation` before calling `setRotationMode`?

File: frontend/device/input.lua [hunk 2]
```diff
@@ -38,6 +38,15 @@ local ABS_MT_POSITION_X = 53
 local ABS_MT_POSITION_Y = 54
 local ABS_MT_TRACKING_ID = 57
 local ABS_MT_PRESSURE = 58
+
+-- For device orientation events (ABS.code)
+
+local ABS_PRESSURE = 24
```
Comment (houqp) [inline frontend/device/input.lua]: I am surprised they use pressure key code to send orientation change event, heh

File: frontend/device/gesturedetector.lua [hunk 3]
```diff
@@ -674,6 +674,44 @@ function GestureDetector:adjustGesCoordinate(ges)
...
+                ges.direction = "northeast"
+            end
+            if ges.relative then
+                ges.relative.x, ges.relative.y = -ges.relative.x, -ges.relative.y
+            end
+        elseif ges.ges == "pinch" or ges.ges == "spread"
+                or ges.ges == "inward_pan"
+                or ges.ges == "outward_pan" then
+            if ges.direction == "horizontal" then
+                ges.direction = "horizontal"
```
Comment (houqp) [inline frontend/device/gesturedetector.lua]: This is not changing anything?

File: frontend/device/generic/device.lua [hunk 4]
```diff
@@ -211,4 +211,10 @@ function Device:retrieveNetworkInfo()
     end
 end
 
+function Device:onRotation()
```
Comment (houqp) [inline frontend/device/generic/device.lua]: How about moving this into UIManager?

**Comment by houqp** [comment]
Great work overall! Thanks for the patch. Also please clean up the commit history a little bit. The easiest way is to checkout a new branch based on the current master and cherry-pick the commit 06002e6. Then do a force push to update this PR branch :)

**Comment by vincent-inspectionapps** [comment]
No worries, will clean up the messes tonight. :)

File: frontend/device/input.lua (hunk 2 above)
Comment (Hzj-jie) [inline frontend/device/input.lua]: +1, and also cannot quite understand why they restrict the usage of page up / down button on voyage.

**Pull request closed without merging**